ir/type.cpp

```diff
@@ -222,27 +256,60 @@ expr FloatType::getDummyValue() const {
 }
 
 expr FloatType::getTypeConstraints() const {
-  // TODO
-  return false;
+  if (defined) {
+    return true;
+  } else {
+    auto bw = sizeVar();
+    // TODO: support more fp types
+    auto isSingle = bw == expr::mkUInt(Single, var_bw_bits);
+    auto isDouble = bw == expr::mkUInt(Double, var_bw_bits);
+    return (isSingle || isDouble);
+  }
 }
 
 expr FloatType::operator==(const FloatType &rhs) const {
-  // TODO
-  return false;
+  return sizeVar() == rhs.sizeVar();
 }
 
 expr FloatType::sameType(const FloatType &rhs) const {
   return true;
 }
 
 void FloatType::fixup(const Model &m) {
-  // TODO
+  if (!defined)
+    fpType = (FpType) m.getUInt(sizeVar());
```

Comment:
please add an assertion to ensure the type belongs to the enum, like in the int class.